### PR TITLE
Enable apps for managed event types & other team event types

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR does the following:
- Enable apps for managed event types
- Share credentials for team event types without having the event type owner present in the booking

*I did not make changes to Rainbow as it is now depreciated*

For apps like ga4 & Fathom, there are no access tokens that are needed; we can assume any team admins can enable these apps for event types types. Although for Stripe since an access token is needed we use the team owner's credentials for all event types.

### Follow up
- For video apps, all team members still need to have the app installed. I think we should give the option of using a single video credential for all bookings created under the team event type or allow team members to set it themselves 
- When handling Stripe credentials a lot of them are hard coded for Stripe. Should we handle payment credentials under the `PaymentService`?
- Tests should be made under each app folder to ensure that this behaviour is acting as expected

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How should this be tested?

- See enabled apps on team event types (not including Stripe)
  - Create a team and invite another admin
  - Create a managed event type
  - Install and enable some or all of the event type apps except Stripe & Rainbow
  - Now, visit that event type as the other team admin. Even if the admin does not have the apps installed, they should still see them enabled on the team event type.
  - Still, as the admin go to one of the user's event types. The uninstalled apps should not appear on the event type apps page.
  - In the managed event type, assign the event type to a team member
  - Under the child event type go to the apps tab. The apps should not be editable.

- Enable Stripe for team events
    - Create any team event as the team owner
    - Install Stripe
    - Enable Stripe and set the payment option to collect money on booking 
    - Assign the event type to members that are not the owner
    - Create a booking. The payment should go through without the team owner present in the booking
    - Repeat the above steps for no-show fee bookings
    - Visit the event type as a team admin. You should not be able to edit any Stripe fields
    - As the team owner, uninstall Stripe. It should disappear from any team events

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -1951,10 +1945,37 @@ async function handler(
...
+          },
+        },
+        select: {
+          members: true,
+        },
+      });
+      const teamOwner = teamMembers?.members.find((member) => member.role === "OWNER");
+
+      if (teamOwner) {
+        paymentCredentialUserId = teamOwner.userId;
```
Comment (joeauyeung) [inline packages/features/bookings/lib/handleNewBooking.ts:1967]: If the booking comes from a team event type and Stripe is enabled, then we should use the team owner's Stripe credentials. If not found then throw an error.

File: packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts [hunk 2]
```diff
@@ -305,6 +305,52 @@ export const deleteCredentialHandler = async ({ ctx, input }: DeleteCredentialOp
     }
   }
 
+  // Check if the user is a team owner and disable stripe for the team event type
```
Comment (joeauyeung) [inline packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts:308]: If the user does not have Stripe installed, it also will not appear on the event type apps page.

File: packages/trpc/server/routers/viewer/apps/getEventTypeApps.handler.ts
Comment (joeauyeung) [inline packages/trpc/server/routers/viewer/apps/getEventTypeApps.handler.ts:1]: This is the procedure that determines which apps to display on the event type apps page.

File: packages/trpc/server/routers/viewer/payments/chargeCard.handler.ts [hunk 3]
```diff
@@ -79,9 +79,38 @@ export const chargeCardHandler = async ({ ctx, input }: ChargeCardHandlerOptions
     },
   };
 
+  let paymentCredentialUserId = ctx.user.id;
+
+  const eventType = booking.eventType;
+  const isTeamEvent = eventType?.schedulingType || eventType?.parentId;
+
+  if (isTeamEvent) {
```
Comment (joeauyeung) [inline packages/trpc/server/routers/viewer/payments/chargeCard.handler.ts:87]: When charging an attendee's card using Stripe, we need to get the team owner's Stripe credentials

**Review by spatecon — APPROVED** [review]

File: apps/web/components/eventtype/EventTypeSingleLayout.tsx [hunk 4]
```diff
@@ -210,7 +209,7 @@ function EventTypeSingleLayout({
     }
     if (isManagedEventType || isChildrenManagedEventType) {
       // Removing apps and workflows for manageg event types by admins v1
-      navigation = navigation.slice(0, -2);
+      navigation = navigation.slice(0, -1);
```
[the remark this answers is not in the document]
Comment (joeauyeung) [reply apps/web/components/eventtype/EventTypeSingleLayout.tsx:212]: Removing this as apps and workflows are now enabled for managed event types

**Comment by leog** [comment]
@joeauyeung I will be focusing on testing this tomorrow, sorry for the delay, workflows for MET took longer than expected.

I think you can focus on adding tests for this in the meantime, as we discussed not adding functionality without tests anymore 😅

BTW, are we still getting pressure to have this implemented a with the quick solution instead of the other sustainable long-term solution we discussed? I still think it should be the way to go, considering Orgs work.

**Comment by joeauyeung** [comment]
Closing this PR as the time urgency is gone. We will use parts of this PR and its ideas when implementing organizations.

**Pull request closed without merging**